src/codegen/statements.rs

```diff
@@ -421,22 +421,10 @@ pub(crate) fn statement(
             if body_reachable {
                 cfg.set_basic_block(next_block);
 
-                if !next.is_empty() {
-                    for stmt in next {
-                        statement(
-                            stmt,
-                            func,
-                            cfg,
-                            contract_no,
-                            ns,
-                            vartab,
-                            loops,
-                            placeholder,
-                            return_override,
-                            opt,
-                        );
-                        body_reachable = stmt.reachable();
-                    }
+                if let Some(next) = next {
+                    expression(next, cfg, contract_no, Some(func), ns, vartab, opt);
+
+                    body_reachable = next.ty() != Type::Unreachable;
                 }
 
                 if body_reachable {
```

Comment:
Looks like this could be entangled a bit:

```rust
                if let Some(next) = next {
                    expression(next, cfg, contract_no, Some(func), ns, vartab, opt);

                   if next.ty() != Type::Unreachable {
                       cfg.add(vartab, Instr::Branch { block: body_block });
                   }
                }
```

Reply:
We still want the branch even if there is no `next`

Reply:
Ah yes you a right